pkg/cache/clusterqueue_test.go

```diff
@@ -1274,3 +1274,144 @@ func TestCohortLendable(t *testing.T) {
 		t.Errorf("Unexpected cohort lendable (-want,+got):\n%s", diff)
 	}
 }
+
+func TestClusterQueueReadinessWithTAS(t *testing.T) {
```

Comment:
This looks great, we only use the exported functions in this test which makes it much more reliable than `TestClusterQueueUpdateWithAdmissionCheck`. Could you try to migrate the `TestClusterQueueUpdateWithAdmissionCheck` into this one? And rename `TestClusterQueueReadinessWithTAS` to just `TestClusterQueueReadiness`. `enableTAS` could be a paratemer

Reply:
`TestClusterQueueReadiness` already exists, so maybe we could merge this one there and simply make the `TestClusterQueueUpdateWithAdmissionCheck` to use exported functions.

Reply:
I would just merge them under `TestClusterQueueReadiness` name. Look at the asserts there: https://github.com/kubernetes-sigs/kueue/pull/3541/files#diff-e11e73ca517b5546e51925b6e6299ff79841efda37d39243ccee269d731de277R597-R598. The intention of the other test seems to test what this test does.